packages/driver/src/cypress/log.ts

```diff
@@ -624,13 +598,53 @@ class LogManager {
         }
       }
 
-      // set the log on the command
-      const current = state('current')
+      let command = log.get('command')
 
-      if (current) {
-        current.log(log)
+      if (!command) {
+        command = state('current')
+        log.set('command', command)
       }
 
+      if (command) {
+        // See commands/asserting.ts for a detailed explanation of commandLogId.
+        // Short version: If onBeforeLog sets the same commandLogId as an existing log on the current command,
+        // then we want to merge the current instance into the existing one, rather than display a new log to
+        // the user.
+        let commandLogId = log.get('commandLogId')
+
+        if (commandLogId != null) {
+          const previousLogInstance = command.get('logs').find(_.matchesProperty('attributes.commandLogId', commandLogId))
+
+          if (previousLogInstance) {
+            // log.merge unsets any keys that aren't set on the new log instance. We
+            // copy over 'snapshots' beforehand so that existing snapshots aren't lost
+            // in the merge operation.
+            log.set('snapshots', previousLogInstance.get('snapshots'))
+            previousLogInstance.merge(log)
+
+            if (previousLogInstance.get('end')) {
+              previousLogInstance.end()
+            }
+
+            return
+          }
+        }
+
+        command.log(log)
+      }
+
+      // if snapshot was passed
+      // in, go ahead and snapshot
+      if (log.get('snapshot') && !log.get('snapshots')) {
```

Comment:
```suggestion
      if (log.get('snapshot')) {
```

Reply:
Ah, yep. This was a remnant from earlier attempts to rein in the number of duplicate snapshots, but no longer necessary now that previousLogInstance logic is working correctly.